src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go

```diff
@@ -302,7 +317,41 @@ func (s *Service) queryFeed(feedURL string) (*gofeed.Feed, []gofeed.Item, error)
 	return feed, items, nil
 }
 
+func containsAny(item string, list []string) bool {
+	for _, i := range list {
+		if strings.Contains(item, i) {
+			fmt.Println("contains", item, i)
+			return true
+		}
+	}
+	return false
+}
+
+func itemFiltered(i *gofeed.Item, mustInclude, mustNotInclude *includeRules) bool {
+	// If `must_include` terms have been specified, the feed item must include at least one item for each field
+	// that has been specified. This means that if both an Author and Title substring have been specified,
+	// both the Author and Title must contain these substrings or the item will be skipped.
```

Comment:
So basically you're ANDing the terms instead of ORing them, which makes sense. This should probably be situated all in one place (I would propose `includeRules` struct) so the reader can fully see the algorithm you wish to apply, without having to hunt down random functions in this file.

Reply:
I put some more descriptive comments in all places (but mostly on `includeRules`). Let me know if you think something is not explained well enough.